dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,383 @@
+            for s in r.subkeys():
+                device_info = {}
+                config = None
+
+                for field in fields:
+                    try:
+                        value = s.value(field).value
+                    except RegistryValueNotFoundError:
+                        continue
+
+                    if field == "NetCfgInstanceId":
+                        if isinstance(value, str):
+                            device_info["NetCfgInstanceId"] = value
+                            config = self._extract_network_device_config(value)
+
+                            if config:
+                                device_info["ip"] = config["ip"]
+                                device_info["dns"] = config["dns"]
+                                device_info["gateway"] = config["gateway"]
+                                device_info["subnetmask"] = config["subnetmask"]
+                                device_info["domain"] = config["domain"]
+                                device_info["metric"] = config["metric"]
+
+                            device_info["Name"] = self._extract_network_device_name(value)
+
+                    elif field == "NetworkAddress":
+                        if isinstance(value, str):
+                            device_info["NetworkAddress"] = value
+
+                    elif field == "NetworkInterfaceInstallTimestamp":
+                        if isinstance(value, int):
+                            unix_timestamp = (value / 10000000) - EPOCH_DIFF
+                            device_info["NetworkInterfaceInstallTimestamp"] = unix_timestamp
+
+                    elif field == "DriverDesc":
+                        if isinstance(value, str):
+                            device_info["DriverDesc"] = value
+
+                    elif field == "*IfType":
+                        if isinstance(value, int):
+                            device_info["Type"] = IfTypes.get(int(value), "Unknown")
+
+                    elif field == "*PriorityVlanTag":
+                        if isinstance(value, str):
+                            device_info["Vlan"] = PriorityVlanTagEnum.get(int(value), "Unknown")
+
+                if "NetCfgInstanceId" in device_info and "Type" in device_info and config:
+                    yield device_info
+
+    def _extract_network_device_config(self, interface_ids) -> list[dict, str]:
+        key = f"HKLM\\SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters\\Interfaces\\{interface_ids}"
+        fields = [
+            "DhcpDefaultGateway",
+            "DefaultGateway",
+            "DhcpIPAddress",
+            "IPAddress",
+            "DhcpNameServer",
+            "NameServer",
+            "DhcpSubnetMask",
+            "SubnetMask",
+            "DhcpDomain",
+            "Domain",
+            "InterfaceMetric",
+        ]
+        config = {"domain": [], "gateway": [], "ip": [], "dns": [], "subnetmask": [], "metric": []}
+
+        for field in fields:
```

Comment:
Same comment here about looping through the fields. I'd hardcode it. If you're doing this for re-use of the `try/except` construction for the registry stuff, throw that in a function. #778 would improve this later down the road.